src/main/java/Gandalf.java

```diff
@@ -0,0 +1,183 @@
+                    }
+                    else if (text.startsWith("deadline ")) {
+                        if (!text.contains("/by")) {
+                            System.out.println("Invalid command.");
+                            System.out.println(LINE);
+                        }
+                        else {
+                            String removeDeadlineString = text.replaceFirst("deadline", "").trim();
+                            String[] parts = removeDeadlineString.split("/by");
+                            String deadlineItem = parts[0];
+                            String dueBy = parts[1];
+                            listTasks[insertIndex] = new Deadline(deadlineItem, dueBy);
+                            System.out.println(LINE);
+                            System.out.println("Got it. I've added this task:");
+                            System.out.println("  " + listTasks[insertIndex]);
+                            insertIndex += 1;
+                            System.out.println("Now you have " + insertIndex + " tasks in the list.");
+                            System.out.println(LINE);
+                        }
+                    }
+                    else if (text.startsWith("event ")) {
+                        if (!text.contains("/from") || !text.contains("/to")) {
+                            System.out.println("Invalid command.");
+                            System.out.println(LINE);
+                        }
+                        else {
+                            String removeEventString = text.replaceFirst("event", "").trim();
+                            String[] firstPartition = removeEventString.split("/from");
+                            String eventItem = firstPartition[0];
+                            String fromAndToString = firstPartition[1];
+                            String[] secondPartition = fromAndToString.split("/to");
+                            String eventFrom = secondPartition[0];
+                            String eventTo = secondPartition[1];
+
+                            listTasks[insertIndex] = new Event (eventItem, eventFrom, eventTo);
+                            System.out.println(LINE);
+                            System.out.println("Got it. I've added this task:");
+                            System.out.println("  " + listTasks[insertIndex]);
+                            insertIndex += 1;
+                            System.out.println("Now you have " + insertIndex + " tasks in the list.");
+                            System.out.println(LINE);
+                        }
+                    }
+                    else {
+                        System.out.println("Invalid keyword.");
+                        System.out.println(LINE);
+                    }
+
+                } else {
+                    System.out.println(LINE);
+                    System.out.println("List is full. Cannot add more items.");
+                    System.out.println(LINE);
```

Comment:
Creating a Printing class that can be inherited by other classes can be a potential solution to reudce code duplication.